app/grandchallenge/components/views.py

```diff
@@ -472,3 +494,114 @@ def get(self, request, *args, **kwargs):
+        help_text = request.GET.get("help-text")
+        current_value = request.GET.get("current-value")
+
+        if widget_choice == FileWidgetChoices.FILE_SEARCH.name:
+            html_content = render_to_string(
+                FileSearchWidget.template_name,
+                {
+                    "widget": FileSearchWidget().get_context(
+                        name=interface,
+                        value=None,
+                        attrs={
+                            "help_text": help_text if help_text else None,
+                        },
+                    )["widget"],
+                },
+            )
+            return HttpResponse(html_content)
+        elif widget_choice == FileWidgetChoices.FILE_UPLOAD.name:
+            html_content = render_to_string(
+                UserUploadSingleWidget.template_name,
+                {
+                    "widget": UserUploadSingleWidget().get_context(
+                        name=interface,
+                        value=None,
+                        attrs={
+                            "id": interface,
+                            "help_text": _join_with_br(
+                                help_text if help_text else None,
+                                file_upload_text,
+                            ),
+                        },
+                    )["widget"],
+                },
+            )
+            return HttpResponse(html_content)
+        elif (
+            widget_choice == FileWidgetChoices.FILE_SELECTED.name
+            and current_value
+            and (
+                ComponentInterfaceValue.objects.filter(
+                    pk=current_value
+                ).exists()
+                if current_value.isdigit()
+                else UserUpload.objects.filter(pk=current_value).exists()
+            )
+        ):
+            # this can happen on the display set update view or redisplay of
+            # form upon validation, where one of the options is the current
+            # image, this enables switching back from one of the above widgets
+            # to the chosen image. This make sure the form element with the
+            # right name is available on resubmission.
+            html_content = render_to_string(
+                HiddenInput.template_name,
+                {
+                    "widget": {
+                        "name": interface,
+                        "value": current_value,
+                        "type": "hidden",
+                    },
+                },
+            )
+            return HttpResponse(html_content)
+        elif widget_choice == FileWidgetChoices.UNDEFINED.name:
+            # this happens when switching back from one of the
+            # above widgets to the "Choose data source" option
+            return HttpResponse()
+        else:
+            raise RuntimeError("Unknown widget type")
+
+
+class FileSearchResultView(LoginRequiredMixin, ListView):

```

Comment:
Thinking about permissions, currently, anyone can visit this view. I think it should be locked down to users who have access to (any) files on GC, similar to how we do it for `FileUploadFormFieldView`.